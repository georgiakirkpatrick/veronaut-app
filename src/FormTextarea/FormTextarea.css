.FormTextarea {
    width: 100%;
}

.FormTextarea textarea {
    margin: 5px 0px 10px;
    padding: 10px;
    font: inherit;
    width: 100%;
    border-radius: 5px;
}

.FormTextarea textarea:focus {
    height: 10vh;
}