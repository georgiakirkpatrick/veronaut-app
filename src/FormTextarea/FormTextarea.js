import React from 'react'
import './FormTextarea.css'

const FormTextarea = props => {
    const {
        currentValue,
        handleChange,
        id,
        name,
        prompt
    } = props

    return (
        <div className='FormTextarea'>
            <label htmlFor={id}>{prompt}</label>
            <textarea
                required
                id={id}
                name={name}
                onChange={handleChange}
                rows='1'
                value={currentValue}
            />
        </div>
    )
}

FormTextarea.defaultProps = {
    currentValue: 1,
    handleChange: () => {},
    id: 1,
    name: '',
    prompt: ''
}

export default FormTextarea