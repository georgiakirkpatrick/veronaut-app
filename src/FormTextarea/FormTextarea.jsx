import React from 'react'
import './FormTextarea.css'

const FormTextarea = ({
  currentValue = 1,
  handleChange = () => {},
  id = 1,
  name = '',
  prompt = ''
}) => {
  return (
    <div className='FormTextarea'>
      <label htmlFor={id}>{prompt}</label>
      
      <textarea
        required
        id={id}
        name={name}
        onChange={handleChange}
        rows='1'
        value={currentValue}
      />
    </div>
  )
}

export default FormTextarea