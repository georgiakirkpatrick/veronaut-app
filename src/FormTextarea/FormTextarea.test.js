import React from 'react'
import ReactDOM from 'react-dom'
import FormTextarea from './FormTextarea'
import renderer from 'react-test-renderer'

describe ('FormTextarea', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FormTextarea />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<FormTextarea />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})