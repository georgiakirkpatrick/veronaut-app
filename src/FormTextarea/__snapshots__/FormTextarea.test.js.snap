// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextarea renders the UI as expected 1`] = `
<div
  className="FormTextarea"
>
  <label
    htmlFor={1}
  >
    
  </label>
  <textarea
    id={1}
    name=""
    onChange={[Function]}
    required={true}
    rows="1"
    value={1}
  />
</div>
`;
